corehq/apps/receiverwrapper/views.py

```diff
@@ -61,11 +63,39 @@
 PROFILE_PROBABILITY = float(os.getenv('COMMCARE_PROFILE_SUBMISSION_PROBABILITY', 0))
 PROFILE_LIMIT = os.getenv('COMMCARE_PROFILE_SUBMISSION_LIMIT')
 PROFILE_LIMIT = int(PROFILE_LIMIT) if PROFILE_LIMIT is not None else 1
+CACHE_EXPIRY_30_DAYS_IN_SECS = 30 * 24 * 60 * 60
+
+
+def _audit_request_auth_errors(user_id, request):
+    cache_key = f"form_submission_audit:{user_id}"
+    if cache.get(cache_key):
+        # User is already logged once in last 30 days for incorrect access, so no need to log again
+        return []
+
+    stack = inspect.stack()
+    function_names = [frame.function for frame in stack if not inspect.isbuiltin(frame.function)]
+
+    def invalid_user_permission():
+        if hasattr(request, 'user'):
+            return not request.user.has_perm('access_mobile_endpoints')
+        return True
+
+    if post_api.__name__ not in function_names and invalid_user_permission():
+        cache.set(cache_key, True, CACHE_EXPIRY_30_DAYS_IN_SECS)
+        return f"Request not made from {post_api.__name__} handler for user"
+
+    return []
```

Comment:
@akashkj 
this function has a mix type of returns, an empty list, boolean and a string.
Can it be simplified to return None or a string?